pico/tools/src/main/java/io/helidon/pico/tools/DefaultInterceptorCreator.java

```diff
@@ -0,0 +1,983 @@
+        return result;
+    }
+
+    private static IdAndToString toDecl(InterceptedElement method) {
+        MethodElementInfo mi = method.elementInfo();
+        String name = (mi.elementKind() == ElementInfo.ElementKind.CONSTRUCTOR) ? CTOR_ALIAS : mi.elementName();
+        String builder = typeNameElementNameAnnotations(mi);
+        return new IdAndToString(name, builder);
+    }
+
+    private static String typeNameElementNameAnnotations(ElementInfo ei) {
+        StringBuilder builder = new StringBuilder(".typeName(create(" + ei.elementTypeName() + ".class))");
+        builder.append("\n\t\t\t.elementName(").append(CodeGenUtils.elementNameRef(ei.elementName())).append(")");
+        TreeSet<AnnotationAndValue> sortedAnnotations = new TreeSet<>(ei.annotations());
+        for (AnnotationAndValue anno : sortedAnnotations) {
+            builder.append("\n\t\t\t.addAnnotation(").append(toDecl(anno)).append(")");
+        }
+        return builder.toString();
+    }
+
+    private static IdAndToString toDecl(ElementInfo elementInfo) {
+        String name = elementInfo.elementName();
+        return new IdAndToString(name, elementInfo.elementTypeName() + " " + name);
+    }
+
+    private static IdAndToString toDecl(AnnotationAndValue anno) {
+        StringBuilder builder = new StringBuilder("DefaultAnnotationAndValue.create(" + anno.typeName() + ".class");
+        Map<String, String> map = anno.values();
+        String val = anno.value().orElse(null);
+        if (map != null && !map.isEmpty()) {
+            builder.append(", Map.of(");
+            int count = 0;
+            TreeMap<String, String> sortedMap = new TreeMap<>(map);
+            for (Map.Entry<String, String> e : sortedMap.entrySet()) {
+                if (count++ > 0) {
+                    builder.append(", ");
+                }
+                builder.append("\"")
+                        .append(e.getKey())
+                        .append("\", \"")
+                        .append(e.getValue())
+                        .append("\"");
+            }
+            builder.append(")");
+        } else if (val != null) {
+            builder.append(", \"")
+                    .append(val)
+                    .append("\"");
+        }
+        builder.append(")");
+        return new IdAndToString(anno.typeName().name(), builder);
+    }
+
+    @SuppressWarnings("checkstyle:OperatorWrap")
+    private static InterceptedMethodCodeGen toBody(InterceptedElement method) {
+        MethodElementInfo mi = method.elementInfo();
+        String name = (mi.elementKind() == ElementInfo.ElementKind.CONSTRUCTOR) ? CTOR_ALIAS : mi.elementName();
+        StringBuilder builder = new StringBuilder();
+        builder.append("public ").append(mi.elementTypeName()).append(" ").append(mi.elementName()).append("(");
+        String args = CommonUtils.toString(mi.parameterInfo().stream().map(ElementInfo::elementName)
+                                     .collect(Collectors.toList()));
```

Comment:
Collection -> stream -> List -> stream -> String?

Reply:
see the mapping in the middle?  Want to extract the element name, and then make them all into a comma-delimited list.

Reply:
I do, please use collector directly:
```suggestion
        String args = mi.parameterInfo().stream().map(ElementInfo::elementName).collect(Collectors.joining(","));
```

Reply:
next push.